src/server/routes/activities.js

```diff
@@ -262,5 +269,82 @@ export default [
       notes: 'Retrieve an activity from the database by its ID',
       tags: ['api', 'activities']
     }
+  },
+  {
+    method: 'DELETE',
+    path: '/activities/{activityId}',
+    options: {
+      /**
+       * Delete an activity by ID
+       *
+       * @param {import('@hapi/hapi').Request request - The Hapi request object
+       *     @param {string} request.params.activityId - The activity id
+       * @param {import('@hapi/hapi').ResponseToolkit} h - The Hapi response toolkit
+       * @returns {Promise<import('@hapi/hapi').ResponseObject>} - A response indicating success or failure
+       */
+      handler: async (request, h) => {
+        try {
+          const activityId = request.params.activityId
+
+          // Begin transaction for atomic operation
+          const transaction = await sequelize.transaction()
+
+          try {
+            // Find IDs of all SmallCatch records associated with the Activity
+            const smallCatchIds = (
+              await SmallCatch.findAll({
+                attributes: ['id'],
+                where: { activity_id: activityId },
+                transaction
+              })
+            )?.map((smallCatch) => smallCatch.id)
+
+            // Delete associated SmallCatchCount
+            await SmallCatchCount.destroy({
+              where: { small_catch_id: smallCatchIds },
+              transaction
+            })
+
+            // Delete associated SmallCatches
+            await SmallCatch.destroy({
+              where: { activity_id: activityId },
+              transaction
+            })
+
+            // Delete associated Catches
+            await Catch.destroy({
+              where: { activity_id: activityId },
+              transaction
+            })
+
+            // Delete the Activity
+            const deletedCount = await Activity.destroy({
+              where: { id: activityId },
+              transaction
+            })
+
+            if (deletedCount === 0) {
+              await transaction.rollback()
+              return handleNotFound(
+                `Activity with ID ${activityId} not found`,
+                h
+              )
+            }
+
+            // Commit transaction
+            await transaction.commit()
+            return h.response().code(StatusCodes.NO_CONTENT)
+          } catch (error) {
+            await transaction.rollback()
+            throw error
+          }
+        } catch (error) {
+          return handleServerError('Error deleting activity', error, h)
```

Comment:
I'm not convinced by having a try...catch nested inside another try...catch like this. Could this line not replace 339 `throw error`, and thereby dispose of the outer try...catch?